src/blocking.rs

```diff
@@ -0,0 +1,325 @@
+//!
+//! A blocking version of the SDK.
+//!
+//! These synchronous functions will block the current thread to execute instead
+//! of returning futures that need to be executed on a runtime. In every other way,
+//! they are identitical to their asynchronous counterparts.
+//!
+//! # Optional
+//! This requires the optional `blocking` feature to be enabled.
+pub use crate::internal::{
+    document_api::{
+        AssociationType, DocAccessEditErr, DocumentAccessResult, DocumentDecryptResult,
+        DocumentDecryptUnmanagedResult, DocumentEncryptResult, DocumentEncryptUnmanagedResult,
+        DocumentId, DocumentListMeta, DocumentListResult, DocumentMetadataResult, DocumentName,
+        UserOrGroup, VisibleGroup, VisibleUser,
+    },
+    group_api::{
+        GroupAccessEditErr, GroupAccessEditResult, GroupCreateResult, GroupGetResult, GroupId,
+        GroupListResult, GroupMetaResult, GroupName, GroupUpdatePrivateKeyResult,
+    },
+    user_api::{
+        DeviceId, EncryptedPrivateKey, UserCreateResult, UserDevice, UserDeviceListResult, UserId,
+        UserResult, UserUpdatePrivateKeyResult,
+    },
+};
+use crate::{
+    document::{advanced::DocumentAdvancedOps, DocumentEncryptOpts, DocumentOps},
+    group::{GroupCreateOpts, GroupOps},
+    user::{DeviceCreateOpts, UserCreateOpts, UserOps},
+    DeviceContext,
+    InitAndRotationCheck::{self, NoRotationNeeded, RotationNeeded},
+    IronOxide, PrivateKeyRotationCheckResult, PublicKey, Result,
+};
+use futures::executor::block_on;
+use std::collections::HashMap;
+
+/// Struct that is used to make authenticated requests to the IronCore API. Instantiated with the details
+/// of an account's various ids, device, and signing keys. Once instantiated all operations will be
+/// performed in the context of the account provided. Identical to IronOxide but also contains a Runtime.
+pub struct BlockingIronOxide {
+    pub(crate) ironoxide: IronOxide,
+    pub(crate) runtime: tokio::runtime::Runtime,
+}
+
+impl BlockingIronOxide {
+    /// Get the `DeviceContext` instance that was used to create this SDK instance
+    pub fn device(&self) -> &DeviceContext {
+        &self.ironoxide.device
+    }
+
+    /// Rotate the private key of the calling user and all groups they are an administrator of where needs_rotation is true.
+    /// Note that this function has the potential to take much longer than other functions, as rotation will be done
+    /// individually on each user/group. If rotation is only needed for a specific group, it is strongly recommended
+    /// to call [user_rotate_private_key()](user\/trait.UserOps.html#tymethod.user_rotate_private_key) or
+    /// [group_rotate_private_key()](group\/trait.GroupOps.html#tymethod.group_rotate_private_key) instead.
+    /// # Arguments
+    /// - `rotations` - PrivateKeyRotationCheckResult that holds all users and groups to be rotated
+    /// - `password` - Password to unlock the current user's user master key
+    pub fn rotate_all(
```

Comment:
Should this link to IronOxide's `rotate_all` for docs?

Reply:
Yes, it should. Fixed.
I have checked that the links work, and I think they look reasonable. 
Example:
![image](https://user-images.githubusercontent.com/34786857/72292965-b35f5480-360f-11ea-99db-616b35ad9b27.png)